### PR TITLE
Complete java future right away if sources are empty

### PR DESCRIPTION
**Pull request by jvican**

### CODE REVIEW

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by bloopoid** [comment]
Performance test failed:

<details>
  <summary>Benchmark logs (last 200 lines)</summary>

</details>

Error line number: 122

[check /tmp/tmp.2dpgiKqfh1/2018-09-24-115108-benchmarks-log.txt for more information]

File: backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopHighLevelCompiler.scala [hunk 1]
```diff
@@ -217,19 +217,29 @@ final class BloopHighLevelCompiler(
...
        * See https://github.com/sbt/zinc/issues/234. */
       if (setup.order == CompileOrder.JavaThenScala) {
-        fireJavaCompilation.flatMap {
-          case JavaSignal.ContinueCompilation => compileJava.flatMap(_ => compileScala)
-          case JavaSignal.FailFastCompilation(failedProjects) =>
-            throw new StopPipelining(failedProjects)
-        }
-      } else {
-        compileScala.flatMap { _ =>
+        if (javaSources.isEmpty) {
```
Comment (tues) [inline backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopHighLevelCompiler.scala]: I think you could avoid code duplication if this `if` was the outer one, as they both seem to be the same (except that `()` argument which I believe is a mistake). WDYT?

Comment (jvican) [reply backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopHighLevelCompiler.scala]: Yeah, you're right :wink:

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Comment by jvican** [comment]
test performance please

**Comment by bloopoid** [comment]
performance test scheduled: 1 job(s) in queue, 0 running.

**Comment by bloopoid** [comment]
Performance test finished successfully.

Benchmarks is based on merging with master

**Pull request merged**